primitives/parachain-inherent/src/client_side.rs

```diff
@@ -99,6 +108,25 @@ async fn collect_relay_storage_proof(
 	relevant_keys.push(relay_well_known_keys::CURRENT_SLOT.to_vec());
 	relevant_keys.push(relay_well_known_keys::ACTIVE_CONFIG.to_vec());
 	relevant_keys.push(relay_well_known_keys::dmq_mqc_head(para_id));
+
+	tracing::debug!(
+		target: LOG_TARGET,
+		?dmp_message_window,
+		"seting dmq_mqc_head_for_message relevant keys",
+	);
+
+	// Collect proof for a subset of messages (`downward_messages_count`).
+	if let Some(first_message) = dmp_message_window.first() {
+		let first_message = first_message.message_idx;
+		let last_message = first_message.wrapping_add(downward_messages_count.into());
+
+		for idx in first_message.into()..last_message.into() {
```

Comment:
if last_message < first_message, the for loop won't run
`last_message` is actually not inclusive, so the naming is a bit confusing